server/journeys/referral/contributory-factors-comments/controller.ts

```diff
@@ -0,0 +1,24 @@
+import { Request, Response } from 'express'
+import { escapeHTML } from '../../../utils/utils'
+
+export class ReferralContributoryFactorsCommentsController {
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const contributoryFactorRows = req.journeyData.referral!.contributoryFactors!.map(factor => [
+      {
+        text: factor.factorType.description,
+      },
+      {
+        classes: 'govuk-link',
+        html: `<a href="${escapeHTML(factor.factorType.code.toLowerCase())}-comment">${escapeHTML(factor.comment) ? 'Edit comment' : 'Add comment'}<span class="govuk-visually-hidden"> for ${escapeHTML(factor.factorType.description)} factor</span></a>`,
```

Comment:
There's a built in ` | escape` function you can use if you do this in the view rather than from the controller which I would recommend. This is a view concern rather than controller logic. Have a look at the [appointments extra information view](https://github.com/ministryofjustice/hmpps-activities-management/blob/main/server/views/pages/appointments/create-and-edit/appointment-set/extra-information.njk#L38) as that is what this pattern is based on. That also used a table rather than a summary list